src/App.tsx

```diff
@@ -1,60 +1,98 @@
-import classNames from 'classnames';
-
-import 'bulma/css/bulma.css';
-import '@fortawesome/fontawesome-free/css/all.css';
+import { useState } from 'react';
+import cn from 'classnames';
 import './App.scss';
 
+import { User, Post } from './types';
+import { getPosts } from './services';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [hasPostsLoadingError, setHasPostsLoadingError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const showNoPostsYet =
+    !!selectedUser && !isLoading && !hasPostsLoadingError && !posts.length;
 
-              <Loader />
+  const showPostList =
```

Comment:
Consider using names like `isPostVisible` or `shouldShowPosts` so it's clear that it's boolean 
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig